ballerina/schemaClient.bal

```diff
@@ -0,0 +1,300 @@
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/sql;
+
+# Represents an SQL metadata client.
+isolated client class SchemaClient {
+    private final Client dbClient;
+    private final string database;
+
+    # Initializes the SchemaClient object
+    #
+    # + host - The name of where the mysql server is hosted (ex: localhost)
+    # + user - The username to access the database
+    # + password - The password to access the database
+    # + database - The name of the database to be accessed
+    # + return - A `sql:Error` or nil
+    public function init(string host, string user, string password, string database) returns sql:Error? {
+        self.database = database;
+        self.dbClient = check new (host, user, password);
+    }
+
+    # Retrieves all tables in the database.
+    #
+    # + return - A string array containing the names of the tables or an `sql:Error`
+    isolated remote function listTables() returns string[]|sql:Error {
+        string[] tables = [];
+        stream<record {|string table_name;|}, sql:Error?> results = self.dbClient->query(
+            `SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES 
+             WHERE TABLE_SCHEMA = ${self.database};`
+        );
+
+        do {
+            check from record {|string table_name;|} result in results
+                do {
+                    tables.push(result.table_name.toString());
+                };
```

Comment:
We can use the `select` query syntax here.
https://github.com/ballerina-platform/module-ballerina-sql/blob/c028f98e214881916f97a9074240730819f99f62/ballerina/tests/schema_client.bal#L34-L35